agent/main.go

```diff
@@ -159,8 +165,8 @@ func main() {
 			},
 		},
 		SkipInstallation: skipInstallation,
```

Comment:
suggestion - we can probably remove this field from `HostReconciler` struct. When doing `installer.New`, you can check for `--skip-installation` flag. If set, then `k8sInstaller` will be `nil`.
In the `Reconcile` function, instead of checking for `SkipInstallation` value, simply check on `k8sInstaller` to be `nil` or not